// @tencent/tnpm/config.js
/**!
 * cnpm - config.js
 *
 * Copyright(c) cnpmjs.org and other contributors.
 * MIT Licensed
 *
 * Authors:
 *  fengmk2 <fengmk2@gmail.com> (http://fengmk2.github.com)
 */

'use strict';

/**
 * Module dependencies.
 */

var path = require('path');
var fs = require('fs');
var program = require('commander');

var root;
if (process.platform === 'win32') {
  root = process.env.USERPROFILE || process.env.APPDATA || process.env.TMP || process.env.TEMP;
} else {
  root = process.env.HOME || process.env.TMPDIR || '/tmp';
}

var config = module.exports = {
  cnpmHost: 'http://tnpm.oa.com',
  cnpmRegistry: 'http://r.tnpm.oa.com',
  disturl: 'http://tnpm.oa.com/mirrors/node', // download dist tarball for node-gyp
  iojsDisturl: 'http://tnpm.oa.com/mirrors/iojs',
  mirrorsUrl: 'http://tnpm.oa.com/mirrors',
  taobaoNpmCdnUrl: 'http://tnpm.oa.com/mirrors/cdn.npm.taobao.org',
  cache: path.join(root, '.tnpm'),  //cache folder name
  userconfig: path.join(root, '.tnpmrc'),
  tnpmcustomconfig: path.join(root, '.tnpmcustom'),
};
